tests/framework/utils.py

```diff
@@ -401,20 +405,22 @@ def run_cmd(cmd, check=False, shell=True, cwd=None, timeout=None) -> CommandRetu
         proc.kill()
         stdout, stderr = proc.communicate()
 
-        # Log the message with one call so that multiple statuses
-        # don't get mixed up
-        CMDLOG.warning(
-            "Timeout executing command: %s\n",
-            _format_output_message(proc, stdout, stderr),
-        )
+        if warn_on_fail:
+            # Log the message with one call so that multiple statuses
+            # don't get mixed up
+            CMDLOG.warning(
+                "Timeout executing command: %s\n",
+                _format_output_message(proc, stdout, stderr),
+            )
 
         raise
```

Comment:
Maybe instead of warning we should attach proc, stdout and stderr to the exception and raise that. The caller can choose to suppress it or display it.

Reply:
I like the explicit logs better because they always get printed in full, while exception messages get truncated by pytest

Reply:
Could we change the logging level in the caller? 

```
prev_level = CMDLOG.getLevel()
CMDLOG.getLevel(logging.ERROR)
...
CMDLOG.setLevel(prev_level)
```

I imagine another option is to use the `warnings.warn`, but that would not get printed the same as with `logging`.

Reply:
Would that work with the parallel tests? But it's a lot more code than this, what advantage do you see in having the caller do it like this instead of just passing a flag? :o